src/seedlink/plugins/seismicpi_plugin/setup_configuration/UPDATE.pl

```diff
@@ -0,0 +1,192 @@
+$net = (split(/=/,$net_line[0]))[1];
+@comp_line = grep(/^COMPONENT/i, @conf);
+$comp = (split(/=/,$comp_line[0]))[1];
+@channel_line = grep(/^CHANNEL/i, @conf);
+$channel = (split(/=/,$channel_line[0]))[1];
+@loc_line = grep(/^LOCATION/i, @conf);
+$loc = (split(/=/,$loc_line[0]))[1];
+@gain_line = grep(/^GAIN/i, @conf);
+$gain = (split(/=/,$gain_line[0]))[1];
+@mode_line = grep(/^MODE/i, @conf);
+$mode = (split(/=/,$mode_line[0]))[1];
+@src_line = grep(/^SOURCE/i, @conf);
+$src = (split(/=/,$src_line[0]))[1];
+
+# Validate input values
+# NET
+$lnet = length $net;
+if ($lnet > 2) { print "\n*** Error: Network length maximum length must be 2 characters, e.g UK, exiting. ***\n"; exit; }
+# STATION
+$lstat = length $station;
+if ($lstat > 5) { print "\n*** Error: Station name maximum length must be 4 characters, e.g TEST, exiting. ***\n"; exit; }
+# COMP
+$lcomp = length $comp;
+if ($lcomp > 1) { print "\n*** Error: Component maximum length must be 1 character, e.g Z, exiting. ***\n"; exit; }
+# CHANNEL 
+$lchan = length $channel;
+if ($lchan > 2) { print "\n*** Error: Channel maximum length must be 2 character2, e.g BH, exiting. ***\n"; exit; }
+# LOCATION 
+$lloc = length $loc;
+if ($lloc > 2) { print "\n*** Error: Location maximum length must be 1 character, e.g 00, exiting. ***\n"; exit; }
+# GAIN
+if ($gain eq "1") { $gain_msg = "0.064 μV/count";}
+elsif ($gain eq "2") { $gain_msg = "0.32 μV/count";}
+elsif ($gain eq "4") { $gain_msg = "0.16 μV/count";}
+elsif ($gain eq "8") { $gain_msg = "0.08 μV/count";}
+else { print "\n*** Wrong gain value defined.\n
+--- For SeismicPi HAT can be 1, 2, 4 or 8:
+	   ‘1’: ×1 = 0.64μV/count 
+           ‘2’: ×2 = 0.32μV/count 
+	   ‘4’: ×2 = 0.16μV/count 
+	   ‘8’: ×2 = 0.08μV/count, exiting. ***\n "; exit; }
+# MODE
+if ($mode eq "1") {$mode_msg = "Single Channel Mode"; }
+elsif ($mode eq "2") {$mode_msg = "Multi Channel Mode"; }
+else { print "\n*** Wrong Channel Mode defined.\n
+--- For SeismicPI HAT can be 1 or 2:
+	   ‘1’: Single Channel Mode
+	   ‘2’: Multi Channel Mode, exiting. ***\n"; exit; }
+# SOURCE
+if ($src eq "1") {$src_msg = "Internal Accelerometer";}
+elsif ($src eq "2") {$src_msg = "External Inputs"; }
+else { print "\n*** Wrong Source defined,\n	
+--- For SeismicPI HAT can be 1 or 2:
+	   ‘1’: Use Internal Accelerometer
+	   ‘2’: Use external inputs, exiting. ***\n"; exit; }
+
+#print "$station, $net, $comp, $channel, $loc, $gain, $mode, $src\n";
+# Configure SeisComP files
+# Uncomment the following to clean old key and config file
+#print "Cleaning old files...\n";
+#`rm /home/sysop/seiscomp3/etc/key/station*`;
+# Make new key file
+$sc3_key = "/home/sysop/seiscomp3/etc/key/station\_$net\_$station";
+open(SC3,"+>$sc3_key") or die "Cannot open $sc3_key\n";
+printf SC3 "seedlink:seismicpi";
+close(SC3);
+print "$sc3_key\n";
+`seiscomp update-config`;
+# Make plugin configuration file
+$plugin_cfg = "/home/sysop/seiscomp3/var/lib/seedlink/$net.$station.ini";
```

Comment:
See above about `$SEISCOMP_ROOT`.

Reply:
I replaced UPDATE.pl with seismicpi_setup.pl and i added to check the $SEICOMP_ROOT variable and if is set to proceed.
I also added a used input for deleting old station config files (seismicpi can work only with one station configuration).